### PR TITLE
restructure workspace

### PR DESCRIPTION
**Pull request by zetanumbers**
Allow examples and any other psp crate to share compiler artifacts to reduce build time and disk usage. `cargo-psp` is excluded since use different rustc parameters for psp and native builds, triggering a whole rebuild when native and psp builds are interleaved.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
-target
-Cargo.lock
+/target
```
Comment (sajattack) [inline .gitignore:1]: My knowledge regarding gitignore is slightly sketchy, but I think
`target/` would match both `/target` and `/cargo-psp/target`

Comment (ZetaNumbers) [reply .gitignore]: i made it just `target`

**Comment by ZetaNumbers** [comment]
Any more issues with this PR?

**Comment by sajattack** [comment]
@overdrivenpotato kick the CI please.

**Pull request merged**